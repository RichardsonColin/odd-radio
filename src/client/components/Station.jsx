import React, { Component } from 'react';
import PlayerButtons from './PlayerButtons.jsx';
import { findStationExpandInfo } from '../util/ClientFunctions.jsx';

class Station extends Component {

  constructor(props) {
    super(props);
    this.state = {
      details: {
        id: this.props.id,
        name: this.props.name,
        audioFeed: this.props.audioFeed,
        streamType: this.props.streamType
      },
      expanded: false
    }

    this.toggleStationInfo = this.toggleStationInfo.bind(this);
  }

  // Switch between expanded true and false for showing info.
  toggleStationInfo() {
    if (this.state.expanded) {
      this.setState({
        expanded: false
      })
    } else {
      this.setState({
        expanded: true
      })
    }
  }

  render() {

    // If it is not expanded, render just some details.
    if (!this.state.expanded) {
      return (
        <div id={ this.props.name }>
          <div className={`container station-container${this.props.stationType}`}>
            <div className="row station-row border">
              <div className="one-third column station-name center"> {this.props.name} </div>
              <div className="one-third column station-play-button center">
                <PlayerButtons clickFunction={this.props.handleSelectedStation} params={this.state.details}
                  streamLoading={this.props.streamLoading} playState={ this.props.playState }
                  activeStation={this.props.activeStation} />
              </div>
              <div className="one-third column station-info-button center">
                <i className="fa fa-chevron-down card-chevron" aria-hidden="true" onClick={(e) => {this.toggleStationInfo(); findStationExpandInfo(this.props.name)}}></i>
              </div>
            </div>
          </div>
        </div>
      )

    // If it is expanded, render all the details.
    } else {
      return (
        <div id={ this.props.name }>
          <div className={ `container station-container${this.props.stationType} info-expanded` }>
            <div className="row station-row border">
              <div className="one-third column station-name center"> </div>
              <div className="one-third column station-play-button center">
                <PlayerButtons clickFunction={ this.props.handleSelectedStation } params={ this.state.details }
                streamLoading={this.props.streamLoading} playState={ this.props.playState }
                activeStation={ this.props.activeStation } />
              </div>
              <div className="one-third column station-info-button center">
                <i className="fa fa-chevron-up card-chevron" aria-hidden="true" onClick={ (e) => {this.toggleStationInfo(); findStationExpandInfo(this.props.name)}}></i>
              </div>
            </div>
          </div>
          <div className="container info-container info-expanded" >
            <div className="row center-align">
              <div className="station-info center">
                <div className="station-branding one-third column"> { this.props.name } </div>
                <div className="two-thirds column info">
                  <div><b>Location:</b> { this.props.city } </div>
                  <div><b>Description:</b> { this.props.description } </div>
                  <div><b>Home Page:</b> <a href={this.props.homePage} target='_blank'> { this.props.homePage } </a> </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      );
    }
  }
}

export default Station;