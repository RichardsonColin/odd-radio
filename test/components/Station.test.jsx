import React from 'react';
import renderer from 'react-test-renderer';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

// React 16 Enzyme adapter
Enzyme.configure({ adapter: new Adapter() });

import Station from '../../src/client/components/Station';

describe('Station', () => {
  const props = {
    key: '1',
    id: '1',
    name: 'QWER',
    description: 'radio station is what we are',
    audioFeed: 'www.google.ca',
    type: 'audio/mp3',
    homePage: 'www.example.com',
    city: 'Saskitoba',
    handleSelectedStation: jest.fn(),
    stationType: 3
  }

  const subject = <Station { ...props }/>;

  it('matches snapshot', () => {
    expect(renderer.create(subject)).toMatchSnapshot();
  });

  describe('onClick', () => {
    const wrapper = shallow(subject);

    it('calls a function', () => {
      const onInfoSelect = spyOn(wrapper.instance(), 'onInfoSelect');
      wrapper.find('.fa-chevron-down').simulate('click');
      expect(onInfoSelect).toHaveBeenCalled();
    });


    // it('changes the chevron to fa-chevron-up which matches the snapshot', () => {
    //   const onInfoSelect = spyOn(wrapper.instance(), 'onInfoSelect');
    //   wrapper.find('.fa-chevron-down').simulate('click');

    //   expect(renderer.create(subject)).toMatchSnapshot();
    // });
  });




});