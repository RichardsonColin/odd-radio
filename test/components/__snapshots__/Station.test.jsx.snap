// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Station matches snapshot 1`] = `
<div
  id="QWER"
>
  <div
    className="container station-container3"
  >
    <div
      className="row station-row border"
    >
      <div
        className="one-third column station-name center"
      >
         
        QWER
         
      </div>
      <div
        className="one-third column station-play-button center"
      >
        <div>
          <i
            className="fa fa-play fa-2x"
            id="pButton"
            onClick={[Function]}
          />
        </div>
      </div>
      <div
        className="one-third column station-info-button center"
      >
        <i
          aria-hidden="true"
          className="fa fa-chevron-down card-chevron"
          onClick={[Function]}
        />
      </div>
    </div>
  </div>
</div>
`;
